chain/store/rand.go

```diff
@@ -79,16 +69,27 @@ func (cs *ChainStore) GetBeaconRandomness(ctx context.Context, blks []cid.Cid, p
 	return DrawRandomness(be.Data, pers, round, entropy)
 }
 
-func (cs *ChainStore) GetChainRandomnessLookingBack(ctx context.Context, blks []cid.Cid, pers crypto.DomainSeparationTag, round abi.ChainEpoch, entropy []byte) ([]byte, error) {
-	return cs.GetChainRandomness(ctx, blks, pers, round, entropy, true)
-}
+func (cs *ChainStore) GetBeaconRandomnessLookingForward(ctx context.Context, blks []cid.Cid, pers crypto.DomainSeparationTag, round abi.ChainEpoch, entropy []byte) ([]byte, error) {
+	randTs, err := cs.GetBeaconRandomnessTipset(ctx, blks, pers, round, entropy, false)
+	if err != nil {
+		return nil, err
+	}
 
-func (cs *ChainStore) GetChainRandomnessLookingForward(ctx context.Context, blks []cid.Cid, pers crypto.DomainSeparationTag, round abi.ChainEpoch, entropy []byte) ([]byte, error) {
-	return cs.GetChainRandomness(ctx, blks, pers, round, entropy, false)
+	be, err := cs.ExtractBeaconEntryForRound(randTs, uint64(round))
+	if err != nil {
+		log.Errorf("failed to get beacon entry as expected: %w", err)
+		// If this failed for some reason, just use the latest entry (lest we halt the chain)
```

Comment:
Bad idea? Better to just fail & stop?

Reply:
If a node can't retrieve the beacon at this round for network reason, that means he'll try to get the one before, is that right ? In that case, it can cause a split with him and the rest of the chain no ?

Reply:
Yes...but if every node runs into the same problem, this would keep the network going (violating the spec in the process)

I'm almost certain we don't wanna do this, but wanted to get feedback

Reply:
wait, does this create a world where a block at an epoch could have two different beacons? if yes, then this is a problem.

Reply:
Yes the default behaviour should be that we can't validate this block (yet), we can't assume "everyone on the network has the same problem accessing drand".

Reply:
This is outdated, the code has been removed